tests/test_time.py

```diff
@@ -58,6 +58,51 @@ def test_map_years(self):
         )
         assert np.array_equal(years, expected)
 
+    def test_map_year_to_data(self):
+        cal = AdventCalendar(anchor_date=(12, 31), freq="180d")
+        # single year in input data
+        time_index = pd.date_range('20210401', '20211231', freq='180d')
+        test_data = np.random.random(len(time_index))
+        timeseries = pd.Series(test_data, index=time_index)
+        year = cal.map_year_to_data(timeseries)
+
+        expected = np.array(
+            [
+                interval("2021-07-04", "2021-12-31"),
+                interval("2021-01-05", "2021-07-04"),
```

Comment:
Note that this last interval is half empty, as your test data starts only in april. I think the package should warn the user about this.

Reply:
Thanks for the inspiration! A warning is added when parts of the generated intervals couldn't be covered by the input data.